Cosminis/WebAPI/Controllers/FriendsController.cs

```diff
@@ -0,0 +1,157 @@
+        }	
+    }
+
+    public IResult SearchByRelationshipId(int relationshipId)
+    {
+    	try
+    	{
+    		Friends relationship = _friendServices.SearchByRelationshipId(relationshipId);
+    		return Results.Ok(relationship); 
+    	}
+    	catch(ResourceNotFound)
+        {
+            return Results.BadRequest("This relationship does not exist."); 
+        }	
+    }
+
+    public IResult ViewRelationshipsByStatus(string status)
+    {
+    	try
+    	{
+    		List<Friends> friendsList = _friendServices.ViewRelationShipsByStatus(status);
+    		return Results.Ok(friendsList); 
+    	}
+    	catch(ResourceNotFound)
+        {
+            return Results.BadRequest("Something went very wrong here."); 
+        }	
+    }
+
+    public IResult CheckRelationshipStatusByUserId(int searchingId, string status)
+    {
+    	try
+    	{
+    		List<Friends> friendsList = _friendServices.CheckRelationshipStatusByUserId(searchingId, status);
+    		return Results.Ok(friendsList); 
+    	}
+    	catch(ResourceNotFound)
+        {
+            return Results.BadRequest("This user has no relationships with this status."); 
+        }	
+    }  
+
+    public IResult FriendsByUserIds(int searchingUserId, int user2BeSearchedFor)
+    {
+    	try
+    	{
+    		Friends friendsReturn = _friendServices.FriendsByUserIds(searchingUserId, user2BeSearchedFor);
+    		return Results.Ok(friendsReturn); 
+    	}
+    	catch(ResourceNotFound)
+        {
+            return Results.BadRequest("These users have no established relationship."); 
+        }	
+    }
+
+    public IResult EditStatus(int editingUserID, int user2BeEdited, string status)
+    {
+    	try
+    	{
+    		Friends editFriendship = _friendServices.EditFriendship(editingUserID, user2BeEdited, status);
+    		return Results.Created("/Friends/EditStatus", editFriendship); 
+    	}
+    	catch(ResourceNotFound)
+        {
+            return Results.BadRequest("These users have no established relationship."); 
+        }	
+    }    
+
+    public IResult CheckRelationshipStatusByUsername(string username, string status)
+    {
+    	try
+    	{
+    		List<Friends> friendsList = _friendServices.CheckRelationshipStatusByUsername(username, status);
+    		return Results.Ok(friendsList); 
+    	}
+    	catch(ResourceNotFound)
+        {
+            return Results.BadRequest("This user has no relationships with this status."); 
+        }	
+    }      
+
+    public IResult AddFriendByUserId(int requesterId, int addedId)
+    {
+    	try
+    	{
+    		Friends newFriends = _friendServices.AddFriendByUserId(requesterId, addedId);
+    		return Results.Created("/Friends/AddFriend", newFriends); 
+    	}
+        catch(ResourceNotFound)
+        {
+            return Results.BadRequest("There is no user with this ID in our system.");
+        }
+    	catch(DuplicateFriends)
+        {
+            return Results.Conflict("You are either friends or never will be!!"); 
```

Comment:
?

Reply:
If people try to add a friend and they are either friends already, blocked, or removed, this will return this exception.